scylla/src/transport/iterator.rs

```diff
@@ -361,6 +385,13 @@ where
                     }
                 };
 
+                // If the query was a driver-side timeout, there is no use
+                // in retrying anymore
+                if matches!(last_error, QueryError::RequestTimeout(_)) {
+                    trace!(parent: &span, "Query timed out driver-side");
+                    break 'nodes_in_plan;
+                }
+
```

Comment:
The idea was that there is a fixed time to fetch a single page. If a query attempt doesn't succeed, the deadline is not reset for the next attempt. Therefore it does not make sense to let the retry policy to decide again in case of driver-side timeout, because the next attempt will already be timed out at the point when it starts - so it's a tiny optimization.

Good point about the inconsistency, I suppose that `execute_query` should get that `if`, too.

Reply:
I guess it makes some sense.

I'm dreaming of the day when we unify iterators and normal queries so that we don't have to everything twice.